test/System/String.test.cpp

```diff
@@ -179,38 +189,38 @@ TEST_CASE("[System] sf::String")
             CHECK(string.toAnsiString() == "\0"s);
             CHECK(string.toWideString() == select(L""s, L"\U0010AFAF"s));
             CHECK(string.toUtf8() == std::basic_string<std::uint8_t>{0xF4, 0x8A, 0xBE, 0xAF});
-            CHECK(string.toUtf16() == std::basic_string<std::uint16_t>{0xDBEB, 0xDFAF});
-            CHECK(string.toUtf32() == std::basic_string<std::uint32_t>{0x10AFAF});
+            CHECK(string.toUtf16() == u"\U0010AFAF"s);
+            CHECK(string.toUtf32() == U"\U0010AFAF"s);
             CHECK(string.getSize() == 1);
             CHECK(!string.isEmpty());
             CHECK(string.getData() != nullptr);
         }
 
         SUBCASE("UTF-32 C string constructor")
         {
-            const sf::String string(reinterpret_cast<const std::uint32_t*>(U"\U0010ABCDrs"));
+            const sf::String string(U"\U0010ABCDrs");
```

Comment:
Since those constructors are implicit, should we use the `const sf::String string = U"...";` syntax?